### PR TITLE
Bicep IaC templates with Streaming Analytics Job Sample (#94)

### PR DESCRIPTION
**Pull request by jmostella**
```
* README.md  Describes how to run the project in Azure and perform functional, unit, and e2e test using an online simulator.
  Added section on "Key Concepts"
* .gitignore Added
* containers.bicep Creates blob storaage container as data sink
* iothubs.bicep Creates IoT Hub
* streamingjobs.bicep Creates ASA job
* main.bicep Creates all Azure resources necessary for sample
* streamanalytics-tech-sample.asaql Sample query for filtering temperature > 27
* test/temperature_equal_to_27_degrees.json Test case data
* temperature_greater_than_27_degrees_expected.json Test case data
* temperature_greater_than_27_degrees.json Test case data
* temperature_less_than_27_degrees.json Test case data
* testConfig.json Added
```
# Type of PR

- [x] Documentation changes
- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose

Adds `Azure Streaming Analytics` Single Tech Sample

## Author pre-publish checklist

- [ ] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [x] Made corresponding changes to the documentation

## Validation steps
Follow markdown documentation

### CODE REVIEW

File: single_tech_samples/streamanalytics/README.md [hunk 1]
```diff
@@ -1,11 +1,138 @@
...
+
+az deployment group create -f main.bicep -g my-rg --parameters query='@./streamanalytics-tech-sample.asaql'
+```
+
+## Functional Test
+
+```bash
+# Add device
+
+az iot hub device-identity create --hub-name iot-tech-sample-test --device-id iot-tech-sample-test --edge-enabled
```
Comment (devlace) [inline single_tech_samples/streamanalytics/README.md:50]: hub-name will have to line up with whatever the user updated their `param name` with. I think its worth to call out explicitly.

Comment (devlace) [reply single_tech_samples/streamanalytics/README.md:50]: Looks like the `az iot` extension is deprecated :|
"This extension 'azure-cli-iot-ext' is deprecated and scheduled for removal. Please remove and add 'azure-iot' instead." --- 
EDIT: aaah, looks like the one I had installed is deprecated, and had to install the azure-iot one (both uses the `az iot` path -- that is confusing :|

**Review by devlace — CHANGES_REQUESTED** [review]
I left some comments -- mostly around the setup scripts. Not for this PR, but I think this would benefit a .devcontainer --probably something for the future. Other than that, its looking good :)

File: single_tech_samples/streamanalytics/README.md [hunk 2]
```diff
@@ -1,11 +1,138 @@
...
+
+```bash
+# Run Unit Tests
+azure-streamanalytics-cicd test -project ./asaproj.json -outputPath ./output/
+
+# Create Resource Group
+az group create -n my-rg -l japaneast
+
+# Validate Generated Template
+az deployment group validate -f main.bicep -g my-rg --parameters query='@./streamanalytics-tech-sample.asaql'
```
[the remark this answers is not in the document]
Comment (jmostella) [reply single_tech_samples/streamanalytics/README.md:31]: For Streaming Jobs setting the compatibility level to `1.2` (recommended) enables automatic partitioning on IoT/Event Hub input.

Unfortunately, it looks like the latest [ARM template schema](https://docs.microsoft.com/en-us/azure/templates/microsoft.streamanalytics/2017-04-01-preview/streamingjobs?tabs=json) only recognizes `1.0` as valid.

There is an enhancement request to suppress warnings in `Bicep`  https://github.com/Azure/bicep/issues/1413 but doesn't look like they have the resources to work on it at this time.

File: single_tech_samples/streamanalytics/README.md (hunk 1 above)
Comment (jmostella) [reply single_tech_samples/streamanalytics/README.md:50]: Good call out here. On further consideration, I think the right thing to do is just remove any confusion and just explicitly set `name` & `env`. And just parameterize all the external commands.

**Review by devlace — APPROVED** [review]

**Pull request merged**